server/src/test/java/io/druid/metadata/TestDerbyDB.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.metadata;
+
+import com.google.api.client.repackaged.com.google.common.base.Throwables;
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import io.druid.common.utils.SocketUtil;
+import io.druid.metadata.storage.derby.DerbyConnector;
+import org.apache.derby.drda.NetworkServerControl;
+import org.junit.rules.ExternalResource;
+
+import java.io.File;
+import java.io.PrintWriter;
+import java.net.InetAddress;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.sql.Connection;
+import java.sql.DriverManager;
+
+public class TestDerbyDB extends ExternalResource
+{
+  private NetworkServerControl networkServerControl = null;
+  private Path tmpPath = null;
+  private int port;
+
+  @Override
+  protected void before() throws Throwable
+  {
+    port = SocketUtil.findOpenPort(1621);
+    tmpPath = Files.createTempFile("derby", ".db");
+    tmpPath.toFile().delete();
```

Comment:
why not just use the existing `TestDerbyConnector`? It uses Derby in-memory, which would remove the need for finding ports, checking connections, and cleaning up files.